### PR TITLE
allow verifying a different name from sni server name

### PR DESCRIPTION
**Pull request by forkner**

### CODE REVIEW

File: tlsdialer_test.go [hunk 1]
```diff
@@ -263,6 +315,16 @@ func TestVariableTimeouts(t *testing.T) {
...
 	time.Sleep(1 * time.Second)
+	// Attempt to clean up and release any blocked goroutines trying to write to this global...
+Cleanup:
+	for true {
+		select {
+		case <-receivedServerNames:
+		default:
+			break Cleanup
+		}
+	}
```
Comment (forkner) [inline tlsdialer_test.go]: sort of a drive-by, this test was kind of gumming up receivedServerNames

File: tlsdialer.go [hunk 2]
```diff
@@ -170,13 +172,23 @@ func (d *Dialer) DialForTimings(network, addr string) (*ConnWithTimings, error)
...
+
 	log.Trace("Copying config so that we can tweak it")
 	configCopy := new(tls.Config)
 	*configCopy = *config
 
 	if d.SendServerName {
-		log.Tracef("Setting ServerName to %s and relying on the usual logic in tls.Conn.Handshake() to do verification", serverName)
 		configCopy.ServerName = serverName
+		if serverName == validateName {
+			log.Tracef("Setting ServerName to %s and relying on the usual logic in tls.Conn.Handshake() to do verification", serverName)
```
Comment (joesis) [inline tlsdialer.go]: Not new to this PR, but I wonder what's the benefit of verifying via stdlib over the DIY approach. The logic is there anyway, and if the code breaks, we want it to break consistently regardless of the server name and validate name.

Comment (forkner) [reply tlsdialer.go]: That's fair... less paths if we just use `verifyServerCerts` in all cases and probably makes any issues with it more obvious.

Comment (forkner) [reply tlsdialer.go]: @joesis updated to always use verifyServerCerts, take another look when you have a chance.

Comment (forkner) [reply tlsdialer.go]: @ox any reason not to here? The only difference according to the docs is that the tls.Conn's VerifiedChains are not populated in this case (you have to use the structure we send back).

It seems more consistent this way anyway since there were some cases where it was not populated in the prior version (and so we likely shouldn't be counting on it)

Comment (oxtoacart) [reply tlsdialer.go]: @forkner My one concern would be that the verification logic doesn't work if we're not explicit about the certs, but it turns out that it does work. I added `TestSimulatedMITMDialingPublicSite` to verify this. So, I think we're good.
